### ISSUE TITLE
[Feature Request] View & Edit Shared Preferences

### ISSUE THREAD

**Issue by mochadwi** [pandulapeter/beagle#67]
Got this references from [Hyperion libraries](https://github.com/willowtreeapps/Hyperion-Android#user-content-core-plugins:~:text=Hyperion%2DShared%2DPreferences%20%2D%20View%20and%20edit%20your%20app's%20key%2Dvalue%20storage.)

cmiiw, does beagle already support this (I didn't find any on the readme)

**Comment by pandulapeter** [comment]
I did consider this at a point, but decided against it for a few reasons.
- [There is a way](https://github.com/pandulapeter/beagle/issues/65) to persist debug flags with Beagle, that's nicely encapsulated and reliable.
- You can display important fields from shared preferences manually in Beagle, for example the [KeyValueListModule](https://github.com/pandulapeter/beagle/blob/master/common/src/main/java/com/pandulapeter/beagle/modules/KeyValueListModule.kt) seems like a good choice. You can also setup custom buttons / switches / sliders to interact with fields that require it.
- Editing a shared preferences file that's not under the library's control is error-prone and hard to generalize (first of all, how to find the file itself? not everybody uses the default name). Also, just because the file is changed, the UI won't be refreshed automatically so that's something to be handled by the developer and people might find that confusing.
- With all the different ways to store data, even simple pairs (now we have EncyptedSharedPreferences and DataStore), I probably couldn't create something that is useful and reliable for every project.

So to summarize, if it's important, it's doable with the current feature set but won't work out of the box.

**Comment by pandulapeter** [comment]
...Plus if I do this I'd have to create a data base inspector as well which is even more complicated, and at that moment there is little reason to compete with the tooling of Android Studio which will always be superior :) Anyway, thanks for the idea!

**Comment by mochadwi** [comment]
Thanks for the explanation @pandulapeter
